runtime/vm/KeyHashTable.c

```diff
@@ -349,6 +349,11 @@ hashClassTableAt(J9ClassLoader *classLoader, U_8 *className, UDATA classNameLeng
 		if (J9ROMCLASS_IS_HIDDEN(clazz->romClass)) {
 			return NULL;
 		}
+		if (J9_ARE_ALL_BITS_SET(clazz->classFlags, J9ClassIsJCLDefineClass | J9ClassIsFrozen)
+			&& J9_ARE_NO_BITS_SET(flags, J9_FINDCLASS_FLAG_JCL_DEFINE_CLASS)
+		) {
```

Comment:
Suggest the (slightly) less expensive test should be first:
```c
		if (J9_ARE_NO_BITS_SET(flags, J9_FINDCLASS_FLAG_JCL_DEFINE_CLASS)
			&& J9_ARE_ALL_BITS_SET(clazz->classFlags, J9ClassIsJCLDefineClass | J9ClassIsFrozen)
		) {
```
Same comment for lines 756-758.